presto-matching/src/test/java/com/facebook/presto/example/ast/RelNode.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.example.ast;
+
+import java.util.List;
+
+public interface RelNode
```

Comment:
These are only available on presto-matching's test classpath, so there's no risk of wrong imports. The test nodes will be shown in 'find class', but not in code completion. They're marked with a green background (test sources) and will have the long package name of `com.facebook.presto.matching.example.rel` - both saying 'those are not the classes you're looking for' :)

In class search the class availability in the classpath of currently open file seems to be taken into account. This means the `rel.example` classes will never show first as long as presto AST is on classpath of currently open file. I've checked that just now.

In spite of the above I believe that shifting the domain of the examples - risking mismatch between what's needed in test domain and production (because we can't mimic production well enough in another domain), and redoing all the tests - is not a justified effort.